federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProvider.java

```diff
@@ -886,7 +887,26 @@ public boolean isConfiguredFor(RealmModel realm, UserModel user, String credenti
     public boolean isValid(RealmModel realm, UserModel user, CredentialInput input) {
         if (!(input instanceof UserCredentialModel)) return false;
         if (input.getType().equals(PasswordCredentialModel.TYPE) && !((UserCredentialManager) user.credentialManager()).isConfiguredLocally(PasswordCredentialModel.TYPE)) {
-            return validPassword(realm, user, input.getChallengeResponse());
+            if(!validPassword(realm, user, input.getChallengeResponse())){
+                return false;
+            }
+
+            // After the password has been validated successfully, check that it still matches the realm's password policy.
+            // Only do that, if we can write the LDAP store, since only then the password can be updated in Keycloak.
+            // TODO: Do we want to support read-only LDAP? How to handle this then? Leave the decision to whoever evaluates the
+            // password policy result, so e.g. show a warning in the browser authenticators / deny authentication?
```

Comment:
How about this?

Reply:
I would propose to enforce the policy and deny authentication in read only mode. If admins don't want that behavior they easily can disable the password policy but they cannot enforce them otherwise. However, there should be a error message to the user imho